protobuf/endpoints/register_function.proto

```diff
@@ -4,8 +4,50 @@ package endpoints;
 
 import "models/registered_function.proto";
 
+// TODO(adrian@preemo.io, 05/10/2023): move some of these to models
+
+message CpuRequirements {
+  // Optional field where 1000 millicores equals 1 CPU.
+  optional uint32 millicores = 1;
+
+  // Optional field indicating amount of memory needed.
+  // Not all precisions may be accepted.
+  // For example, the worker may choose to only accept values that are multiples of Mebibytes.
+  optional uint64 memory_in_bytes = 2;
+
+  // Optional field indicating amount of storage needed.
+  // Not all precisions may be accepted.
+  // For example, the worker may choose to only accept values that are multiples of Mebibytes.
+  optional uint64 storage_in_bytes = 3;
+}
+
+enum GpuModel {
+  GPU_MODEL_UNSPECIFIED = 0;
+  GPU_MODEL_NVIDIA_TESLA_K80 = 1;
```

Comment:
It might be a good time, to figure out which models are we going to support. I put some random stuff mostly for testing purposes, but since this is a public proto, we should think about it a little.

Reply:
ah okay good to know. I'm going to remove this enum for now anyway, but it will be good to document which specific machines are supported. It probably makes sense to just start with a couple (maybe even 1) just as an mvp